packages/eslint-plugin/src/rules/require-localize-metadata.ts

```diff
@@ -0,0 +1,100 @@
+import type { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [
+  {
+    readonly requireDescription?: boolean;
+  },
+];
+
+const DEFAULT_OPTIONS: Options[number] = {
+  requireDescription: false,
+};
+
+const VALID_LOCALIZED_STRING_WITH_DESCRIPTION = new RegExp(
+  /:(.*\|)?([\w\s]+){1}(@@.*)?:.+/,
+);
+
+// TODO: The following values are also used in eslint-plugin-template/i18n, and should likely be centralized
+const STYLE_GUIDE_LINK = 'https://angular.io/guide/i18n';
+const STYLE_GUIDE_LINK_COMMON_PREPARE = `${STYLE_GUIDE_LINK}-common-prepare`;
+const STYLE_GUIDE_LINK_METADATA_FOR_TRANSLATION = `${STYLE_GUIDE_LINK_COMMON_PREPARE}#i18n-metadata-for-translation`;
+
+export type MessageIds = 'requireLocalizeMetadata';
+export const RULE_NAME = 'require-localize-metadata';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that $localize tagged messages contain a description.',
+      recommended: false,
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          requireDescription: {
```

Comment:
I don't understand what purpose this option serves given the whole rule is dedicated to this purpose?

This is further backed up by the fact you don't have tests for when it is disabled.

I would think that if someone does not want to require the description they would simply not use this rule right?

Reply:
@JamesHenry no worries, and I hope things are less chaotic now 😄 . The reason for this set up is to be able to support custom ID and meaning, which are the other two i18n metadata parameters (see [i18n metadata for translation](https://angular.io/guide/i18n-common-prepare#i18n-metadata-for-translation)). This pull request is only to support description for the time being, which is why the implementation looks the way it does (i.e. having functions like `testLocalizeTemplateElementWithDescription`).

With that all said, let me know what you think about the overall approach.